### DISCUSSION TITLE
[description=stream:error, attributes={code=500}]

### DISCUSSION THREAD

**Discussion by MuYi0828** [Auties00/WhatsappWeb4j#248] [General]
Hello, I tried to use the Kotlin language to achieve your project, but I encountered difficulty. I scan the code after generating a QR code, but the answer it gives me is "unable to connect", and the return code is "CODE = 500 ", I can't solve this problem. I can't understand the information this code gives me feedback. I think I will cry immediately.

**Comment by Auties00** [comment]
I've never seen that error, can you show me your code? Also what version are you using?

**Comment by MuYi0828** [comment]
> 我从未见过那个错误，你能给我看你的代码吗？另外，您使用的是哪个版本？

I clone the latest code of your project, and then use the Kotlin language to try to achieve the same features with your code, but after scanning the QR code, whatsApp prompts me to be unable to connect the device for the time being. I tried to find out the problem by exclusion, but I failed. By the way, the WebSocket client is achieved through the library of Java-WebSocket. I don't know if it is for this reason. Or maybe because I used the wrong alternative method in the process of using the Kotlin language, it caused a certain data transmission error.

**Comment by MuYi0828** [comment]
> I've never seen that error, can you show me your code? Also what version are you using?

There is no problem with your code. Of course, this is my bug; in fact, I should not seek answers from you. In fact, I hope that you just know the meaning of the return "Code = 500". Maybe I can try to solve this bug slowly.

**Comment by MuYi0828** [comment]
> 我从未见过那个错误，你能给我看你的代码吗？另外，您使用的是哪个版本？

onOpen: Switching Protocols
digestNode：Node[description=iq, attributes={xmlns=md, from=s.whatsapp.net, id=3045372400, type=set}, content=[Node[description=pair-device, content=[Node[description=ref, content=[50, 64, 71, 49, 67, 111, 57, 75, 71, 113, 117, 90,...........
sendNode: Node[description=iq, attributes={id=3045372400, to=s.whatsapp.net, type=result}]
digestNode：Node[description=stream:error, attributes={code=500}]
digestNode：Node[description=xmlstreamend]
digestNode：Node[description=stream:error, attributes={code=503}]
digestNode：Node[description=xmlstreamend]
onClose: code: 1000, reason:, remote: false

**Comment by Auties00** [comment]
> > 我从未见过那个错误，你能给我看你的代码吗？另外，您使用的是哪个版本？
> 
> I clone the latest code of your project, and then use the Kotlin language to try to achieve the same features with your code, but after scanning the QR code, whatsApp prompts me to be unable to connect the device for the time being. I tried to find out the problem by exclusion, but I failed. By the way, the WebSocket client is achieved through the library of Java-WebSocket. I don't know if it is for this reason. Or maybe because I used the wrong alternative method in the process of using the Kotlin language, it caused a certain data transmission error.

I mean just import the library in your Kotlin project and use whatever you need: there's really no benefit with your approach.

**Comment by Auties00** [comment]
> > 我从未见过那个错误，你能给我看你的代码吗？另外，您使用的是哪个版本？
> 
> onOpen: Switching Protocols
> digestNode：Node[description=iq, attributes={xmlns=md, from=s.whatsapp.net, id=3045372400, type=set}, content=[Node[description=pair-device, content=[Node[description=ref, content=[50, 64, 71, 49, 67, 111, 57, 75, 71, 113, 117, 90,...........
> sendNode: Node[description=iq, attributes={id=3045372400, to=s.whatsapp.net, type=result}]
> digestNode：Node[description=stream:error, attributes={code=500}]
> digestNode：Node[description=xmlstreamend]
> digestNode：Node[description=stream:error, attributes={code=503}]
> digestNode：Node[description=xmlstreamend]
> onClose: code: 1000, reason:, remote: false

Without seeing the full code it's kind of hard to guess the problem, though it's probably a ciphering error

**Comment by MuYi0828** [comment]
> > > 我从未见过那个错误，你能给我看你的代码吗？另外，您使用的是哪个版本？
> > 
> > 
> > I clone the latest code of your project, and then use the Kotlin language to try to achieve the same features with your code, but after scanning the QR code, whatsApp prompts me to be unable to connect the device for the time being. I tried to find out the problem by exclusion, but I failed. By the way, the WebSocket client is achieved through the library of Java-WebSocket. I don't know if it is for this reason. Or maybe because I used the wrong alternative method in the process of using the Kotlin language, it caused a certain data transmission error.
> 
> I mean just import the library in your Kotlin project and use whatever you need: there's really no benefit with your approach.



> > > 我从未见过那个错误，你能给我看你的代码吗？另外，您使用的是哪个版本？
> > 
> > 
> > I clone the latest code of your project, and then use the Kotlin language to try to achieve the same features with your code, but after scanning the QR code, whatsApp prompts me to be unable to connect the device for the time being. I tried to find out the problem by exclusion, but I failed. By the way, the WebSocket client is achieved through the library of Java-WebSocket. I don't know if it is for this reason. Or maybe because I used the wrong alternative method in the process of using the Kotlin language, it caused a certain data transmission error.
> 
> I mean just import the library in your Kotlin project and use whatever you need: there's really no benefit with your approach.

I think you are right. But maybe I can try it again. If I find the problem and successfully solve it, I will share it under this ISSUE. Thank you very much for your answer.

**Comment by MuYi0828** [comment]
> > > 我从未见过那个错误，你能给我看你的代码吗？另外，您使用的是哪个版本？
> > 
> > 
> > onOpen: Switching Protocols
> > digestNode：Node[description=iq, attributes={xmlns=md, from=s.whatsapp.net, id=3045372400, type=set}, content=[Node[description=pair-device, content=[Node[description=ref, content=[50, 64, 71, 49, 67, 111, 57, 75, 71, 113, 117, 90,...........
> > sendNode: Node[description=iq, attributes={id=3045372400, to=s.whatsapp.net, type=result}]
> > digestNode：Node[description=stream:error, attributes={code=500}]
> > digestNode：Node[description=xmlstreamend]
> > digestNode：Node[description=stream:error, attributes={code=503}]
> > digestNode：Node[description=xmlstreamend]
> > onClose: code: 1000, reason:, remote: false
> 
> Without seeing the full code it's kind of hard to guess the problem, though it's probably a ciphering error

I think so too. I should continue to study your code, and then try to modify my code. I should find the problem. There is an old saying in China: as long as the kung fu is deep, the iron pestle is grinded into a needle.

**Comment by JesseSBezerra** [comment]
Estou recebendo esse erro com uma certa frequência, sabe me dizer o que posso fazer para resolver ?:
java.lang.RuntimeException: Node timed out(Node[description=iq, attributes={xmlns=w:sync:app:state, to=s.whatsapp.net, id=ea-13, type=set}, content=[Node[description=sync, content=[Node[description=collection, attributes={name=critical_block, return_snapshot=true, version=0}], Node[description=collection, attributes={name=critical_unblock_low, return_snapshot=true, version=0}], Node[description=collection, attributes={name=regular_high, return_snapshot=true, version=0}], Node[description=collection, attributes={name=regular_low, return_snapshot=true, version=0}], Node[description=collection, attributes={name=regular, return_snapshot=true, version=0}]]]]]), no response from WhatsApp
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:636)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at it.auties.whatsapp.model.request.Request.cancelTimedFuture(Request.java:50)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.RuntimeException: Node timed out(Node[description=iq, attributes={xmlns=w:sync:app:state, to=s.whatsapp.net, id=ea-13, type=set}, content=[Node[description=sync, content=[Node[description=collection, attributes={name=critical_block, return_snapshot=true, version=0}], Node[description=collection, attributes={name=critical_unblock_low, return_snapshot=true, version=0}], Node[description=collection, attributes={name=regular_high, return_snapshot=true, version=0}], Node[description=collection, attributes={name=regular_low, return_snapshot=true, version=0}], Node[description=collection, attributes={name=regular, return_snapshot=true, version=0}]]]]]), no response from WhatsApp
	at it.auties.whatsapp.model.request.Request.<init>(Request.java:38)
